### PR TITLE
NODE: Correct signature for readFileSync

### PR DESCRIPTION
**Pull request by cjbarth**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/fs.html#fs_fs_readfilesync_path_options
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -2834,7 +2834,7 @@ declare module "fs" {
      * @param fileName
      * @param options An object with optional {encoding} and {flag} properties.  If {encoding} is specified, readFileSync returns a string; otherwise it returns a Buffer.
      */
-    export function readFileSync(filename: string, options: { encoding: string; flag?: string; }): string;
+    export function readFileSync(filename: string, options: { encoding?: string | null; flag?: string; }): string;
```
Comment (Flarna) [inline types/node/index.d.ts]: If encoding is not present the return type is a Buffer not a string.

Comment (cjbarth) [inline types/node/index.d.ts]: Good catch. I've updated the definitions to account for this. The problem is that I'm always passing in `encoding`, but sometimes the value is `null`, so I need that to be accounted for in the typings.

File: types/node/index.d.ts [hunk 2]
```diff
@@ -2828,13 +2828,17 @@ declare module "fs" {
      * @param encoding
      */
     export function readFileSync(filename: string, encoding: string): string;
+    export function readFileSync(filename: string, encoding: null): Buffer;
```
Comment (Flarna) [inline types/node/index.d.ts]: As tsconfig.json for node has strictNullChecks set to false the overload with encoding: null needs to be first as the encoding: string variant matches also null.

File: types/node/index.d.ts [hunk 3]
```diff
@@ -2828,13 +2828,17 @@ declare module "fs" {
...
+    export function readFileSync(filename: string, encoding: string | null): string | Buffer;
     /**
      * Synchronous readFile - Synchronously reads the entire contents of a file.
      *
      * @param fileName
      * @param options An object with optional {encoding} and {flag} properties.  If {encoding} is specified, readFileSync returns a string; otherwise it returns a Buffer.
      */
     export function readFileSync(filename: string, options: { encoding: string; flag?: string; }): string;
+    export function readFileSync(filename: string, options: { encoding: null; flag?: string; }): Buffer;
```
Comment (Flarna) [inline types/node/index.d.ts]: same as above, because variant with null needs to be first.

**Review by Flarna — CHANGES_REQUESTED** [review]
Please add tests for the new supported use cases.

**Review by Flarna — APPROVED** [review]

**Pull request merged**